src/QueryLanguage/Form/DataTransformer/StringToExpresionTransformer.php

```diff
@@ -14,7 +14,7 @@ class StringToExpresionTransformer implements DataTransformerInterface
 
     public function __construct(?Grammar $grammar = null)
     {
-        $this->grammar = $grammar ?? new Grammar();
+        $this->grammar = $grammar ?? self::getGrammar();
```

Comment:
Instead of having in this transformer the static method `getGrammar()`, what do you think about making `Grammar` class singleton?
In this way we can have `Grammar::getInstance()`